pkgs/development/tools/replay-io/default.nix

```diff
@@ -0,0 +1,65 @@
+{stdenv, lib, fetchurl, autoPatchelfHook, makeWrapper, libcxx, libX11, libXt, libXdamage, glib, gtk3, dbus-glib, openssl }:
+let
+    metadata = lib.importJSON ./meta.json;
+in
+stdenv.mkDerivation rec {
+    name = "replay-io";
+    version = metadata.date + "-" + metadata.revision;
+    binaryName = "replay";
+    srcs = [
+      (fetchurl metadata.replay)
+      (fetchurl metadata.recordreplay-lib)
+    ];
+    sourceRoot = ".";
+    unpackPhase = ''
+      runHook preUnpack
+
+      arr=($srcs)
+      tar xf ''${arr[0]} --strip-components=1
+      tar xzf ''${arr[1]}
+
+      runHook postUnpack
+    '';
+    buildInputs = [
+        autoPatchelfHook
+        makeWrapper
+        dbus-glib
+        glib
+        gtk3
+        libX11
+        libXdamage
+        libXt
+        libcxx
+        openssl
+    ];
+    patchPhase = ''
+        patchelf --set-interpreter ${stdenv.cc.bintools.dynamicLinker} ${binaryName}
```

Comment:
Are you sure that's actually needed given that you're using `autoPatchelfHook`? If yes, please use `postPatch` rather than `patchPhase`, otherwise stuff like `patches = []` won't work anymore.